src/core/headers.ts

```diff
@@ -151,3 +151,15 @@ export function parseHeaders(
 
   return result;
 }
+
+export function keysToLowercase<T extends Record<string, unknown>>(
+  obj: T,
+): { [K in keyof T as Lowercase<string & K>]: T[K] } {
+  const result: any = {};
+  let key: keyof typeof obj;
+  for (key in obj) {
+    if (Object.prototype.hasOwnProperty.call(obj, key))
+      result[key.toLowerCase()] = obj[key];
+  }
```

Comment:
Do you had any reason to use `in + hasOwnProperty` than `Object.keys`?

Reply:
Just to avoid any inherited keys, but I just changed it to use `Object.entries` which should accomplish the same thing.